src/iconv.jl

```diff
@@ -102,11 +108,11 @@ function iconv_reset!(s::Union{StringEncoder, StringDecoder})
                 (Ptr{Void}, Ptr{Ptr{UInt8}}, Ref{Csize_t}, Ptr{Ptr{UInt8}}, Ref{Csize_t}),
                 s.cd, C_NULL, C_NULL, s.outbufptr, s.outbytesleft)
 
-    if ret == reinterpret(Csize_t, -1)
+    if ret == typemax(Csize_t)
```

Comment:
Ah, yes, that looks like a good choice, so you'd have `-1%Csize_t` then instead of typemax(Csize_t)?  I'll change that then.